src/sagemaker/local/data.py

```diff
@@ -233,8 +235,24 @@ def split(self, file):
         Returns: generator for the individual records that were split from
         the file
         """
-        with open(file, "r") as f:
-            yield f.read()
+        with open(file, "rb") as f:
+            buf = f.read()
+            if not self._is_binary(buf):
+                buf = buf.decode()
+            yield buf
+
+    def _is_binary(self, buf):
+        """binary check.
+
+        binary or text check.
+
+        Args:
+            buf (bytes): bytes in target file.
+
+        Returns:
+            is binary(True) or text(False).
+        """
+        return bool(buf.translate(None, self._textchars))
```

Comment:
please add a comment to explain what the translate function does and why can this decide if a bytearray only contains non-utf8 characters

Reply:
done.

Reply:
0x07-0x0a, 0x0c-0x0d, 0x20-0x7e, 0x80-0xFF

this is not text charactor ranges. 
`open(file, 'r')` error occured charactor.

Reply:
ref: https://stackoverflow.com/questions/898669/how-can-i-detect-if-a-file-is-binary-non-text-in-python
ref: https://stackoverflow.com/questions/32184809/python-file1-why-are-the-numbers-7-8-9-10-12-13-27-and-range0x20-0x100